SWIGGYDEMO_REACT/swiggy-react-app/src/components/itemDetailsAndCart/components/listByCategory/ListbyCategory.js

```diff
@@ -1,162 +1,181 @@
-                    arrayOfVegOnlyItems.push(arrayOfItems);
-                    filterCat.push(i);
-                }
-            })
+import React, { useEffect, useMemo } from 'react';
+import Items from './components/items';
+import { connect } from 'react-redux';
+import updateSearchItemsWithCategoryList from '../../../../actions/searchItemsWithCategoryActionCreator';
+import updateItemsWithCategoryList from '../../../../actions/itemsWithCategoryActionCreator';
+import { menuItemList } from '../../../../models/itemList';
+import { sideMenu } from '../../../../models/menu';
+
+const groupBy = (items) =>
+    items.reduce((cache, item) => {
+        const property = item['category'];
+
+        if (property in cache) {
+            return { ...cache, [property]: cache[property].concat(item) };
+        } else {
+            return { ...cache, [property]: [item] };
         }
-
-        else {
-            category.forEach( (i) => {
-                let arrayOfItems = [];
-                arrayOfItems = data.filter( (d) => {
-                    return d.category === i && (d.name.toUpperCase().indexOf(searchInputText.toUpperCase()) > -1);
-                } )
-                if(arrayOfItems.length != 0) {
-                    arrayOfSearchedItems.push(arrayOfItems);
-                    searchCat.push(i);
+    }, {});
+
+function ListByCategory({
+    itemsWithCategory,
+    searchItemsWithCategory,
+    isVeg,
+    searchInputText,
+    updateSearchItemsWithCategoryList,
+    updateItemsWithCategoryList,
+}) {
+    const { items, categoryList } = itemsWithCategory;
+    const { searchItems, searchCategoryList } = searchItemsWithCategory;
+
+    let itemsDetails = useMemo(() => {
+        return groupBy(menuItemList);
+    }, [menuItemList]);
+    let categoryListForOriginalItems = sideMenu;
+
+    function updateItemsAndCategoryList() {
+        if (isVeg === true) {
+            const updatedItems = {};
+            const categoryList = [];
+            categoryListForOriginalItems.forEach((cat) => {
+                updatedItems[cat] = itemsDetails[cat].filter((item) => {
+                    return item.isVeg === true;
+                });
+                if (updatedItems[cat].length === 0) {
+                    delete updatedItems[cat];
+                } else {
+                    categoryList.push(cat);
                 }
-            })
+            });
+            itemsDetails = updatedItems;
+            categoryListForOriginalItems = categoryList;
+        }
 
-            category.forEach( (i) => {
-                let arrayOfItems = [];
-                arrayOfItems = data.filter( (d) => {
-                    return d.category === i;
-                } )
-                if(arrayOfItems.length != 0) {
-                    arrayOfVegOnlyItems.push(arrayOfItems);
-                    filterCat.push(i);
+        if (searchInputText.length !== 0) {
+            const updatedSearchItems = {};
```

Comment:
use const and map in same line like 

const data = list.map(....);

you can use map in place of forEach.

Reply:
Used const with map